lib/omniauth/strategies/saml.rb

```diff
@@ -186,9 +186,12 @@ def handle_response(raw_response, opts, settings)
         @attributes = response.attributes
         @response_object = response
 
+        # With ruby-saml 1.4.2, that line is never reached with an empty name_id on the response
+        # :nocov:
         if @name_id.nil? || @name_id.empty?
           raise OmniAuth::Strategies::SAML::ValidationError.new("SAML response missing 'name_id'")
         end
+        # :nocov:
```

Comment:
Actually, I think this code can be removed. The ruby-saml is required to be `~> 1.4` and the oldest version matching that tests for the `name_id` presence itself.

Wdyt?